### DISCUSSION TITLE
How to remove white borders on retina displays

### DISCUSSION THREAD

**Discussion by liveman1** [openseadragon/openseadragon#2085] [Q&A] [answered]
Hi,
on retina displays I get white borders in my deep zoom image as shown in the screenshots. They are on each zoom / pyramid level. I do not see them on screens with less resolution.
Probably there are tiny white or soft edges in the generated image tiles.

Is there any way to remove these white borders? E.g. overlapping the tiles by 1 px etc.?

Thanks a lot
![IMG-0382](https://user-images.githubusercontent.com/7196290/148919400-4ec2cfde-227c-458e-9a59-afe79f7f33bd.jpg)
![IMG-0380](https://user-images.githubusercontent.com/7196290/148919402-4f2cb7ff-f761-484c-828c-951ac8bf6ef3.jpg)
![IMG-0381](https://user-images.githubusercontent.com/7196290/148919403-89b48b3e-4c5d-43b2-8e84-8f55e5958c5e.jpg)
![IMG-0379](https://user-images.githubusercontent.com/7196290/148919406-93d581ce-daba-4677-a468-c9b3899204ed.jpg)
!

**Comment by alvaromartmart** [comment]
Probably related to https://github.com/openseadragon/openseadragon/issues/1683, are you using Firefox? Check if you face the same issue in Google Chrome

**Reply by liveman1** [reply]
Hi, 
this happens in all browsers (Safari, Chrome, Firefox) on my iPad.
Thanks a lot for the hint! it is the same issue as #1683, since I use png tiles with transparency. There was a solution currently merged into the master #2075. I will test, if this solves the issue.  

Thanks again!!!

**Comment by liveman1** [comment]
I created a new build from the current master containing #2075 & tested it. I also tried different tile sizes, overlaps etc.
However, I still get the white lines **on retina displays** (iPad, iPhone etc.). There is no issue on low resolution screens.
I guess #1683 was Firefox specific and not related to retina displays. In this issue it was mentioned, that everything worked fine in v2.3.1. This is also the case for my issue. Using 2.3.1 there are no white borders on retina displays.
When taking over the old code in lines 280 - 283 from 2.3.1 in tiles.js on the right to the latest version on the left everything works fine, at least for me.

![SharedScreenshot](https://user-images.githubusercontent.com/7196290/149029155-350d8f14-1532-4fac-8c41-3a3e9e499635.jpg)

Please find a screencast of the issue below to see what I mean.

https://user-images.githubusercontent.com/7196290/149022376-818fb22e-5fcc-43d9-a911-144e5d4cc982.mov

This should be made an issue again.
What do you think?

**Comment by iangilman** [comment, marked as answer]
#2075 doesn't just fix it; you have to set the settings properly to tell OSD how you want it fixed (since the fix has trade-offs). The simplest fix (with the latest master) is to add this to your viewer options: 

```
subPixelRoundingForTransparency: OpenSeadragon.SUBPIXEL_ROUNDING_OCCURRENCES.ALWAYS
```

Clearly this is something that should be documented better, as in, in a place where people would look for it, but I'm not sure the best way to do that. Maybe it's time for us to have an FAQ?

**Reply by liveman1** [reply]
Perfect! Thanks a lot for your help!
It works, using this setting.  :)
